sympy/solvers/solvers.py

```diff
@@ -360,12 +357,12 @@ def checksol(f, symbol, sol=None, **flags):
                 return not nz
             break
 
+        if numerical and val.is_number:
+            return (abs(val.n(18).n(12, chop=True)) < 1e-9) is S.true
         if val == was:
             continue
         elif val.is_Rational:
             return val == 0
-        if numerical and val.is_number:
-            return (abs(val.n(18).n(12, chop=True)) < 1e-9) is S.true
```

Comment:
These `if` conditions have been reordered to prevent the `val==was` condition preventing the numerical check from executing in cases where the equation is a number. Before this change, stuff like `checksol(1e-10, ...)` would return `False` instead of `True` because the numerical check would never execute.